packages/frontend/app/lib/validate.server.ts

```diff
@@ -127,3 +127,30 @@ export const updateWalletAddressSchema = z
     status: z.enum([WalletAddressStatus.Active, WalletAddressStatus.Inactive])
   })
   .merge(uuidSchema)
+
+export const updateTenantSchema = z
+  .object({
+    apiSecret: z
+      .string()
+      .min(10, { message: 'API Secret should be at least 3 characters long' })
+      .max(255, { message: 'Maximum length of API Secret is 255 characters' })
+      .regex(
+        /^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$/,
+        { message: 'API Secret should be Base64 encoded.' }
+      ),
```

Comment:
Do we need it to be? Something we can discuss

Reply:
No, resolved. :-)